src/features.ts

```diff
@@ -24,3 +30,15 @@ export const DefaultFeatureSetForTesting: Features = {
 
 export const FeatureEnabled = (feature: Feature): boolean =>
   feature === Feature.Enabled;
+
+const boolToEnabled = (currentState: boolean): Feature => (currentState ? Feature.Enabled : Feature.Disabled);
+
+// @TODO: build features from settings
```

Comment:
Consolidating these isn't as straightforward as I thought, is it? You may already have the rest of the solution thought out, but two things that might make it easier:

1. No runtime-togglable settings, they require a restart (and we read the applied settings at startup, enabling us to build a static feature set.) This might not be a bad idea to also prevent at least some corner cases of mode switching?

2. Present the feature set as functions rather than values. That way we'd be able to have some settable without compromising the hardcoded set. And still maybe prevent get through the store